rbd/mirror.go

```diff
@@ -71,6 +71,34 @@ func (imm ImageMirrorMode) String() string {
 	}
 }
 
+//  GetMirrorUUID returns a string naming the mirroring uuid for the pool
+//  associated with the ioctx.
+//
+//  Implements:
+//  int rbd_mirror_uuid_get(rados_ioctx_t io_ctx,
+//	                       char *uuid, size_t *max_len);
+func GetMirrorUUID(ioctx *rados.IOContext) (string, error) {
+	var (
+		err   error
+		buf   []byte
+		cSize C.size_t
+	)
+	retry.WithSizes(1024, 1<<16, func(size int) retry.Hint {
```

Comment:
I chose buffer size to be `1024` but I am still confused if I should change it to `512` as given here in the [ceph's pybind](https://github.com/ceph/ceph/blob/77edcfdcdd732ce75987035a46d8c9a560e57b34/src/pybind/rbd/rbd.pyx#L1908).

Reply:
That should be fine. It's not part of the public interface so if we need to we can easily change it later.  I think that as long as it is greater than the size of a string-formatted UUID its going to be fine, but making it 36 or less would be too small and immediately trigger a retry.
In short, this is not an issue. :-)

Reply:
Thanks!.
Will start writing tests now.